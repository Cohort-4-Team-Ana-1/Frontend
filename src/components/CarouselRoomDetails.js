import React from "react";
import Slider from "react-slick";

import "slick-carousel/slick/slick.css";
import "slick-carousel/slick/slick-theme.css";
import { useFetch } from "../hooks/useFetch";

export const CarouselRoomDetails = () => {
  const settings = {
    dots: true,
    infinite: true,
    speed: 500,
    slidesToShow: 1,
    slidesToScroll: 1,
  };

  const data = useFetch();

  return (
    <>
      <div id="slider__container">
        <Slider {...settings}>
          {data[1].map((item) => (
            <img src={item} key={item} alt="room image" />
          ))}
        </Slider>
        <div className="slider__footer">
          <p>$ {new Intl.NumberFormat("es-CO").format(data[0])} COP / Noche</p>
        </div>
      </div>
    </>
  );
};
