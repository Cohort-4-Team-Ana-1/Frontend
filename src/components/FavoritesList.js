import React from 'react'
import { Link } from 'react-router-dom'
import { useFavorites } from '../hooks/useFavorites'


export const FavoritesList = () => {

    const favList = useFavorites()
    console.log(favList)

    return (
        <section className="cards__container">
    
    
    
          { favList &&
          
          
          favList.map((item) => (
            <div className="card" key={`${item._id}`}>
              <img className="card__roomImage" src={item.main_image} alt="" />
              <p className="card__price">
                {new Intl.NumberFormat("es-CO").format(item.price)} COP
              </p>

              <div className="card__info">
                <Link onClick = {()=>{
              sessionStorage.setItem('roomId', `${item._id}`)
            }}
            
            
            to = {`/room-details/${item._id}`} className="card__address">{item.address}</Link>
              </div>
              <img
                className="card__userimage"
                src={item.owner.user_picture}
                alt=""
              />
              <p className="card__username">{item.owner.first_name}</p>
            </div>
          ))}
        </section>
      );
}
