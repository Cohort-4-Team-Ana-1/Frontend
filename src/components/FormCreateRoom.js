import React from "react";
import { useForm } from "react-hook-form";
import axios from "axios";

export const FormCreateRoom = () => {
  const main_image = localStorage.getItem("main_image");
  const secondary_images = JSON.parse(localStorage.getItem("secondary_images"));
  const userId = sessionStorage.getItem("user-id");
<<<<<<< HEAD
  const { register, handleSubmit, errors } = useForm();
  const onSubmit = (data) => {

    const newData = { ...data, main_image, secondary_images, userId };

=======
  const { register, handleSubmit } = useForm();
  const onSubmit = (data) => {
    const newData = { ...data, main_image, secondary_images, userId };
>>>>>>> 4e5dc40248e28dac0fa96c2b9cf3557640bab9f2

    axios({
      url: "/rooms",
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      data: newData,
    })
      .then((response) => {
        alert("Cuarto creado, te redirigimos para que puedas verlo");
        const roomId = response.data.body._id;
        sessionStorage.setItem("roomId", roomId);
        window.location.href = `/room-details/${roomId}`;
      })
      .catch((err) => {
        // console.error(err);
      });
  };

  return (
    <section className="room-section">
      {localStorage.getItem("secondary_images") &&
        localStorage.getItem("main_image") && (
          <form onSubmit={handleSubmit(onSubmit)}>
            <div className="room-basic-data">
              <label className="register-info available">
                <p>Esta disponible:</p>
                <input
                  type="checkbox"
                  placeholder="available"
                  name="available"
                  ref={register}
                  defaultChecked
                />
              </label>
              <label className="register-info">
                <p>Ciudad:</p>
                <select name="city" ref={register({ required: true })}>
                  <option value="Leticia">Leticia</option>

                  <option value="Medellín">Medellín</option>

                  <option value="Arauca">Arauca</option>

                  <option value="Barranquilla">Barranquilla</option>

                  <option value="Cartagena">Cartagena</option>

                  <option value="Tunja">Tunja</option>

                  <option value="Manizales">Manizales</option>

                  <option value="Florencia">Florencia</option>

                  <option value="Yopal">Yopal</option>

                  <option value="Popayán">Popayán</option>

                  <option value="Valledupar">Valledupar</option>

                  <option value="Quibdó">Quibdó</option>

                  <option value="Montería">Montería</option>

                  <option value="Bogotá">Bogotá</option>

                  <option value="Inírida">Inírida</option>

                  <option value="Guaviare Cali">Guaviare Cali</option>

                  <option value="Neiva">Neiva</option>

                  <option value="Rioacha">Rioacha</option>

                  <option value="Santa Marta Cali">Santa Marta Cali</option>

                  <option value="Villavicencio">Villavicencio</option>

                  <option value="Pasto">Pasto</option>

                  <option value="Cúcuta">Cúcuta</option>

                  <option value="Mocoa">Mocoa</option>

                  <option value="Armenia">Armenia</option>

                  <option value="Pereira">Pereira</option>

                  <option value="San Andrés">San Andrés</option>

                  <option value="Bucaramanga">Bucaramanga</option>

                  <option value="Sincelejo">Sincelejo</option>

                  <option value="Ibagué">Ibagué</option>

                  <option value="Cali">Cali</option>

                  <option value="Mitú">Mitú</option>

                  <option value="Puerto Carreño">Puerto Carreño</option>

                  <option value="Medellin">Medellin</option>
                  <option value=" Bogota"> Bogota</option>
                  <option value=" Cali"> Cali</option>
                  <option value=" Barranquilla"> Barranquilla</option>
                </select>
              </label>

              <label className="register-info">
                <p>Título de tu habitación: </p>
                <input
                  type="text"
                  required="required"
                  placeholder="room_name"
                  name="room_name"
                  ref={register({ required: true, min: 3 })}
                />
              </label>

              <label className="register-info">
                <p>Precio del alquiler mensual (COP):</p>
                <input
                  type="number"
                  placeholder="price"
                  name="price"
                  required="required"
                  ref={register}
                />
              </label>

              <label className="register-info">
                <p>Dirección:</p>
                <input
                  type="text"
                  placeholder="address"
                  name="address"
                  ref={register}
                />
              </label>

              <label className="register-info">
                <p>
                  Área del cuarto (mts<sup>2</sup>):
                </p>
                <input
                  type="number"
                  placeholder="square_meters"
                  required="required"
                  name="square_meters"
                  ref={register}
                />
              </label>
            </div>

            <p>Selecciona el mobiliario con el que cuenta el cuarto</p>
            <div className="room-details">
              <label>
                Cama
                <input
                  type="checkbox"
                  placeholder="bed"
                  name="bed"
                  ref={register}
                />
              </label>

              <label>
                Escritorio
                <input
                  type="checkbox"
                  placeholder="desk"
                  name="desk"
                  ref={register}
                />
              </label>

              <label>
                Armario{" "}
                <input
                  type="checkbox"
                  placeholder="closet"
                  name="closet"
                  ref={register}
                />
              </label>

              <label>
                Silla{" "}
                <input
                  type="checkbox"
                  placeholder="chair"
                  name="chair"
                  ref={register}
                />
              </label>

              <label>
                Sofá{" "}
                <input
                  type="checkbox"
                  placeholder="couch"
                  name="couch"
                  ref={register}
                />
              </label>
            </div>

            <p>
              Selecciona los servicios a los que tendrán acceso tus huéspedes
            </p>

            <div className="room-details">
              <label>
                Cocina
                <input
                  type="checkbox"
                  placeholder="kitchen"
                  name="kitchen"
                  ref={register}
                />
              </label>

              <label>
                Visitas
                <input
                  type="checkbox"
                  placeholder="visits"
                  name="visits"
                  ref={register}
                />
              </label>

              <label>
                Internet
                <input
                  type="checkbox"
                  placeholder="wifi"
                  name="wifi"
                  ref={register}
                />
              </label>

              <label>
                Parqueadero
                <input
                  type="checkbox"
                  placeholder="parking"
                  name="parking"
                  ref={register}
                />
              </label>

              <label>
                Lavadora <br />
                <input
                  type="checkbox"
                  placeholder="washing_machine"
                  name="washing_machine"
                  ref={register}
                />
              </label>

              <label>
                Televisión
                <input
                  type="checkbox"
                  placeholder="television"
                  name="television"
                  ref={register}
                />
              </label>

              <label>
                Calefacción
                <input
                  type="checkbox"
                  placeholder="heating"
                  name="heating"
                  ref={register}
                />
              </label>

              <label>
                Baño privado
                <input
                  type="checkbox"
                  placeholder="private_bathroom"
                  name="private_bathroom"
                  ref={register}
                />
              </label>

              <label>
                Gimnasio
                <input
                  type="checkbox"
                  placeholder="gymnasium"
                  name="gymnasium"
                  ref={register}
                />
              </label>

              <label>
                Aire acondicionado
                <input
                  type="checkbox"
                  placeholder="air_conditioner"
                  name="air_conditioner"
                  ref={register}
                />
              </label>
            </div>

            <label className="room-text-info">
              <p>
                Cuéntale a tus huéspedes los beneficios que tiene el vivir
                contigo, habla un poco de ti también.
              </p>
              <textarea
                required="required"
                placeholder="Escribe"
                name="room_description"
                ref={register}
              />
            </label>

            <button type="submit">Guardar</button>
          </form>
        )}
    </section>
  );
};
