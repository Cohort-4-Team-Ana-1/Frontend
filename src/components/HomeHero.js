import React, { useState } from "react";
import { useForm } from "react-hook-form";
import { Link } from "react-router-dom";
import { useFetchSearch } from "../hooks/useFetchSearch";
import void_heart from "../assets/void_heart.svg";
import { RoomListed } from "./RoomListed";
import { useFetchFavorites } from "../hooks/useFetchFavorites";

export const HomeHero = () => {
  const city = JSON.parse(localStorage.getItem("city"));

  const { register, handleSubmit } = useForm();
  const onSubmit = (data) => {
    // console.log(data.search);
    localStorage.setItem("city", JSON.stringify(data.search));
    // window.location.href = `/rooms/${city}`
  };
  //   console.log(errors);

  const search = useFetchSearch(city);
  
  const [fav, setFav] = useState()
console.log(fav)
const  favorites = useFetchFavorites(fav)
console.log(favorites)


  console.log(search);

  console.log("hol");

  return (

    <>
        


      <section className="heroHome">
        <div className="hero__banner">
          <img
            src="https://cdn.pixabay.com/photo/2016/03/09/09/42/buildings-1245953_1280.jpg"
            alt=""
          />
          <div className="hero__input">
            <h1>¡Encuentra tu roomie ideal!</h1>
            <form onSubmit={handleSubmit(onSubmit)}>
              <input
                type="search"
                placeholder="¿En qué ciudad quieres vivir?"
                name="search"
                ref={register}
                autoComplete="off"
              />
              <button type="submit">Buscar</button>
            </form>
          </div>
        </div>
      </section>

    {
        !city ? <RoomListed/>
        : 
            <>
            <h2>Habitaciones disponibles en {city}</h2>
        <section className="cards__container">
          {search !== undefined &&
            search.map((item) => (
                <div className="card" key={`${item._id}`}>
                <img className="card__roomImage" src={item.main_image} alt="" />
                <p className="card__price">
                  {new Intl.NumberFormat("es-CO").format(item.price)} COP
                </p>
                <img src={void_heart} className="card__heart" alt="" onClick = {()=>{
                  
                  //  handleFavorites
                  !sessionStorage.getItem('user-id')? alert('debes tener una cuenta para agregar a favoritos')
                  :localStorage.setItem('favRoom', item._id) && alert('agregado')
                  setFav(localStorage.getItem('favRoom'))
                }
                }/>
                <div className="card__info">
                  <Link onClick = {()=>{
                sessionStorage.setItem('roomId', `${item._id}`)
              }}
              
              
              to = {`/room-details/${item._id}`} className="card__address">{item.address}</Link>
                </div>
                <img
                  className="card__userimage"
                  src={item.owner.user_picture}
                  alt=""
                />
                <p className="card__username">{item.owner.first_name}</p>
              </div>
            ))}
        </section>
  </>




    }


      

    
    </>
  );
};
