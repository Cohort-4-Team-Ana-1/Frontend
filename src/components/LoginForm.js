import React from "react";
import { useForm } from "../hooks/useForm";
import axios from "axios";

export const LoginForm = () => {
  const [formValues, handleInputChange] = useForm({
    email: "",
    password: "",
  });

  const { email, password } = formValues;

  const handleSubmit = (e) => {
    e.preventDefault();
    const data = {
      email: `${email}`,
      password: `${password}`,
    };
    axios({
      url: "/users/sign-in",
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      data: data,
    })
      .then((response) => {
        const datos = response.data;
        sessionStorage.setItem("token", datos.token);
        sessionStorage.setItem("user-id", datos.payload._id);
        sessionStorage.setItem("username", datos.payload.user);
        alert(
          `Hola ${sessionStorage.getItem(
            "username"
          )}, disfruta tu tiempo en Roompali!`
        );
        window.location.href = "/choose-user-rol";
      })
      .catch((err) => {
        // console.error(err);
      });
  };

  return (
    <div>
      <div className="modal-body">
        <h1>Roompali</h1>
        <form action="/" onSubmit={handleSubmit}>
          <h1>Inicia Sesion</h1>
          <div className="form-group">
            <label>
              Correo electronico
              <input
                type="email"
                name="email"
                placeholder="email@gmail.com"
                autoComplete="off"
                value={email}
                onChange={handleInputChange}
              />{" "}
            </label>

            <label>
              Contrase√±a
              <input
                type="password"
                name="password"
                placeholder="*****"
                value={password}
                onChange={handleInputChange}
              />
            </label>
            <button type="submit">Entrar</button>
          </div>
        </form>
      </div>
    </div>
  );
};
