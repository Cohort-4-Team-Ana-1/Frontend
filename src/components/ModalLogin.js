import React, { useState } from "react";
import Modal from "react-modal";

export const ModalLogin = () => {
  const [modalLoginIsOpen, setModalLoginIsOpen] = useState(false);

  return (
    <div>
      <button onClick={() => setModalLoginIsOpen(true)}>
        Open Modal Login
      </button>
      <Modal
        isOpen={modalLoginIsOpen}
        onRequestClose={() => setModalLoginIsOpen(false)}
      >
        <h2>Modal Title Login</h2>
        <p>Modal Body Login</p>
        <div>
          <button onClick={() => setModalLoginIsOpen(false)}>Exit</button>
        </div>
      </Modal>
    </div>
  );
};
