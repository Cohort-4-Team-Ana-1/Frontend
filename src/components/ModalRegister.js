import React, { useState } from "react";
import Modal from "react-modal";

export const ModalRegister = () => {
  const [modalRegisterIsOpen, setModalRegisterIsOpen] = useState(false);

  return (
    <div>
      <button onClick={() => setModalRegisterIsOpen(true)}>
        Open Modal Register
      </button>
      <Modal
        isOpen={modalRegisterIsOpen}
        onRequestClose={() => setModalRegisterIsOpen(false)}
      >
        <div className="modal-body">
          <h2>
            Sign UP at Roompali<span> For Free!</span>
          </h2>
          <form>
            <div>
              <input
                type="text"
                name="name"
                id="name"
                placeholder="First Name"
                required=""
              />
            </div>
            <div>
              <input
                type="email"
                name="email"
                placeholder="E-mail"
                required=""
              />
            </div>
            <div>
              <input
                type="password"
                name="pass"
                className="form-control"
                placeholder="Password"
                required=""
              />
            </div>
            <input id="sign_up" type="button" value="Sign Up" />
          </form>
          <button onClick={() => setModalRegisterIsOpen(false)}>Exit</button>
        </div>
      </Modal>
    </div>
  );
};
