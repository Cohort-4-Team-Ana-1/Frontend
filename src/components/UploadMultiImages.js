import React, { useState } from "react";
import { useForm } from "react-hook-form";
import axios from 'axios'

export const UploadMultiImages = () => {
  const { register, handleSubmit } = useForm();

  const onSubmit = async (data) => {
    const formData = new FormData();
    const toArray = Object.values(data.picture);
    console.log(toArray);
    toArray.map((item) => {
      console.log(item);
      formData.append("image", item);
    });
    axios({
      url: "/images",
      method: "POST",
      data: formData,
    })
      .then((response) => {
        // response.json();
        console.log(response);
        response.data.success === false &&
          alert(
            "Asegurate de que tu archivo sea una imagen con extension .JPG o PNG"
          );

        // window.location.reload(false);
      })
      .catch((err) => {
        console.error(err);
        console.log(err)
      });
  };


  //   const res = await fetch(
  //     "https://backend-roompali.vercel.app/api/images/multi",
  //     {
  //       method: "POST",
  //       body: formData,
  //     }
  //   ).then((res) => res.json());
  //   console.log(res);
  //   res.success === false &&
  //     alert(
  //       "Asegurate de que tus archivos sean imagenes con extension .JPG o PNG"
  //     );
  // };

  const [profileImage, setprofileImage] = useState({
    profileImg:
      "https://images.igdb.com/igdb/image/upload/t_cover_big/co254s.jpg",
  });

  const { profileImg } = profileImage;

  return (
    <>
      <h1> CAPTURA DE IMAGENES MULTIPLES </h1>
      <p>
        Escoge varias imagenes para que la gente conozca tu cuarto a detalle
      </p>
      <form onSubmit={handleSubmit(onSubmit)}>
        <input ref={register} type="file" name="picture" multiple />
        <img
          src={profileImg}
          accept="image/*"
          style={{ width: `100px` }}
          alt=""
        />
        <button>Submit</button>
      </form>
    </>
  );
};
