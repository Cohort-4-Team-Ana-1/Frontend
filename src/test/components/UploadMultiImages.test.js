import { shallow } from "enzyme";
import React from "react";

import "@testing-library/jest-dom";
import "mutationobserver-shim";
import { UploadMultiImages } from "../../components/UploadMultiImages";
global.MutationObserver = window.MutationObserver;

describe("<UploadMultiImages/>", () => {
  const preventDefault = jest.fn();

  const wrapper = shallow(<UploadMultiImages />);
  test('should render UploadMultiImages', () => {
    const wrapper = shallow(<UploadMultiImages />);
    expect(wrapper.length).toEqual(1);
})

  test("should submit", () => {
    wrapper.find("form").simulate("submit", { preventDefault });
    expect(preventDefault).toHaveBeenCalledTimes(1);
    wrapper.unmount();
  });

  test("snapshot to <UploadMultiImages/>", () => {
    const wrapper = shallow(<UploadMultiImages />);

    expect(wrapper).toMatchSnapshot();
  });
});
