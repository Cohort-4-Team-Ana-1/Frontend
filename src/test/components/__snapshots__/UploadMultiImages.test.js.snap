// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UploadMultiImages/> snapshot to <UploadMultiImages/> 1`] = `
<section
  className="upload__form"
>
  <p>
    Escoge varias imagenes para que la gente conozca tu cuarto a detalle
  </p>
  <form
    className="upload__singleImage"
    onSubmit={[Function]}
  >
    <input
      id="multipleImageRoom"
      multiple={true}
      name="picture"
      onChange={[Function]}
      required="required"
      type="file"
    />
    <label
      htmlFor="multipleImageRoom"
    >
      <img
        accept="image/*"
        alt="default image"
        className="choose_image"
        src="default_multiple_pictures.svg"
        style={
          Object {
            "height": "200px",
            "width": "200px",
          }
        }
      />
    </label>
    <button>
      Submit
    </button>
  </form>
</section>
`;
